lib/AST/ASTImporter.cpp

```diff
@@ -1796,7 +1796,16 @@ Error ASTNodeImporter::ImportDefinition(
   if (To->getDefinition() || To->isBeingDefined()) {
     if (Kind == IDK_Everything)
       return ImportDeclContext(From, /*ForceImport=*/true);
```

Comment:
Maybe the `completeDefinition` is needed in this case too?

Reply:
No, because we take this branch when class is not a lambda, so `startDefinition` is called only later, in line 1803.

In case of lambdas, however, `CXXRecordDecl::CreateLambda` behaves as `startDefinition` , i.e it sets the DefinitionData ptr and sets isBeingDefined to true.